flixel/input/gamepad/XboxButtonID.hx

```diff
@@ -125,16 +125,16 @@ class XboxButtonID
 	/**
 	 * Button IDs
 	 */
-	public static inline var A:Int = 10;
-	public static inline var B:Int = 11;
-	public static inline var X:Int = 12;
-	public static inline var Y:Int = 13;
-	public static inline var LB:Int = 8;
-	public static inline var RB:Int = 9;
-	public static inline var BACK:Int = 5;
-	public static inline var START:Int = 4;
-	public static inline var LEFT_ANALOG:Int = 6;
-	public static inline var RIGHT_ANALOG:Int = 7;
+	public static inline var A:Int = 0;
+	public static inline var B:Int = 1;
+	public static inline var X:Int = 2;
+	public static inline var Y:Int = 3;
+	public static inline var LB:Int = 4;
+	public static inline var RB:Int = 5;
+	public static inline var BACK:Int = 6;
+	public static inline var START:Int = 7;
+	public static inline var LEFT_ANALOG:Int = 8;
+	public static inline var RIGHT_ANALOG:Int = 9;
 	
 	public static inline var XBOX:Int = 14;
 	
```

Comment:
So the getters for the dpad directions tell you if something is currently pressed, but they're kind of ugly:

1. They only work on cpp, not flash. So inconsistent.
2. They don't give you justpressed/justreleased, only "is pressed now" or not.

The above "1." is an option if you generate fake dpad button press event ids. Basically you would create a constant like "DPAD_DOWN = 10" and so on, and then just assign those by hand in the function based on whether the hat was moved x = -1/1, or y= -1/1, etc. And make sure you use an unused button constant for what you set DPAD_DOWN/LEFT/RIGHT/UP to. 

This could be wrapped up in a way that is consistent with flash's behavior at the API level.